libwarpaffine/sliceswriter/SlicesWriterTbb.h

```diff
@@ -33,6 +34,9 @@ class CziSlicesWriterTbb : public ICziSlicesWriter
     };
 
     tbb::concurrent_bounded_queue<SubBlockWriteInfo2> queue_;
+    std::map<std::pair<int, int>, libCZI::GUID> retilingIds_;
```

Comment:
I am undecided whether I like this idea or not... I'd phrase it as - we have the newly introduced slice_id (which identifies bricks originating from the same input-bricks), and we have a z-index. Together, they identify "subblocks which are to be grouped into an acquisition-tile and get the same retiling-id". So far, so good.
What I am not completely happy with is the fact that all GUIDs will pile up here (and use memory, an increasing number of CPU-cycles for adding and retrieval, introduce complexity due to concurrency). Granted, we can expect the number of GUID to be of order "several thousand max" and the performance costs are for sure negligible for almost all cases, but still...
So, I wonder if there would be a different approach, avoiding costs and complexity. One idea would be to pass in the GUID (instead of the slice_id), but then we'd have the task of generating it at the call-site (and not sure if it would be easier to do there). Another idea is - the requirement is that the slice_id must be **unique within the CZI-document only** (not globally and for all history), so I wonder - would it be an idea if we construct the GUID in the following way:
* we generate one GUID (e.g. in the c'tor of the slices-writer-object)
* we then overwrite 4 bytes of this GUID with the slice_id, and another 4 bytes with the z-index. 

The resulting GUID then would be composed of 8 "truly random bytes" and 8 bytes containing slide_id and z-index. It would still be "rather unique in the world", and it would surely be guaranteed to be unique in the CZI-document.
What do you think? It would surely be easier and faster, the only downside I can see is that this will not scale if we have "more ids making up retiling-id" (where I wouldn't see the fact that the GUID is now "less unique" as an issue at all).

Reply:
this seems to work. additionally the guids become much more meaningful when inspecting the sub-blocks lateron.